src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -352,18 +352,35 @@ export const useSearch = (services: DiscoverViewServices) => {
   useEffect(() => {
     (async () => {
       const savedSearchInstance = await getSavedSearchById(savedSearchId);
-      setSavedSearch(savedSearchInstance);
 
-      // if saved search does not exist, do not atempt to sync filters and query from savedObject
-      if (!savedSearch) {
-        return;
+      const query =
+        savedSearchInstance.searchSource.getField('query') ||
+        data.query.queryString.getDefaultQuery();
+
+      const isEnhancementsEnabled = await uiSettings.get('query:enhancements:enabled');
+      if (isEnhancementsEnabled && query.dataset) {
+        let pattern = await data.indexPatterns.get(
+          query.dataset.id,
+          query.dataset.type !== 'INDEX_PATTERN'
+        );
+        if (!pattern) {
+          await data.query.queryString.getDatasetService().cacheDataset(query.dataset, {
+            uiSettings: services.uiSettings,
+            savedObjects: services.savedObjects,
+            notifications: services.notifications,
+            http: services.http,
+            data: services.data,
+          });
+          pattern = await data.indexPatterns.get(
```

Comment:
good question.

for anything that isn't an index pattern, it will read just from the cache. if nothing returns from the cache, then we will recreate the the dataset and save it as an index pattern when we call cache dataset. then we retrieve the actual index pattern we just created.

if it is an index pattern it will check the cache if it doesn't exist in the cache it will make it a request